controllers/clusterenvironment_controller.go

```diff
@@ -400,12 +402,59 @@ func (r *ClusterEnvironmentReconciler) updateClusterEnvironmentStatus(ctx contex
 	ce.Status.State = cs.State
 	meta.SetStatusCondition(&ce.Status.Conditions, cs.Condition())
 }
+func (r *ClusterEnvironmentReconciler) getRelatedClusterEnvironments(ctx context.Context, req ctrl.Request, envname string) ([]primazaiov1alpha1.ClusterEnvironment, error) {
+	cee := primazaiov1alpha1.ClusterEnvironmentList{}
+
+	if err := r.List(ctx, &cee, &client.ListOptions{Namespace: req.Namespace}); err != nil {
+		return nil, err
+	}
+
+	ff := r.filterClusterEnvironments(envname, cee.Items)
+	return ff, nil
+}
+func (r *ClusterEnvironmentReconciler) filterClusterEnvironments(
+	environmentName string,
+	clusterEnvironments []primazaiov1alpha1.ClusterEnvironment) []primazaiov1alpha1.ClusterEnvironment {
+
+	cee := []primazaiov1alpha1.ClusterEnvironment{}
+	for _, ce := range clusterEnvironments {
+		if ce.Spec.EnvironmentName == environmentName {
+			cee = append(cee, ce)
+		}
+	}
+
+	return cee
+}
 
-func (r *ClusterEnvironmentReconciler) finalizeClusterEnvironment(ctx context.Context, ce *primazaiov1alpha1.ClusterEnvironment) error {
+func (r *ClusterEnvironmentReconciler) finalizeClusterEnvironment(ctx context.Context, req ctrl.Request, ce *primazaiov1alpha1.ClusterEnvironment) error {
+	var errs []error
```

Comment:
We are not dealing with multiple errors here, so no need for a slice.

Reply:
okay